ghcide/src/Development/IDE/Plugin/Completions.hs

```diff
@@ -151,7 +156,7 @@ getCompletionsLSP ide plId
                 exportsCompls = mempty{anyQualCompls = exportsCompItems}
             let compls = (fst <$> localCompls) <> (fst <$> nonLocalCompls) <> Just exportsCompls
 
-            pure (opts, fmap (,pm,binds) compls, moduleExports)
+            pure (opts, fmap (,pm,binds) compls, moduleExports, lModules)
```

Comment:
Instead of returning `lModules`, I would weave them into the `compls` value reusing the Monoid instance. That way you don't need to add a return value here, nor an extra argument to `getCompletions

Reply:
Much cleaner, thanks!